src/ConfigEditor.tsx

```diff
@@ -10,36 +10,81 @@ interface Props extends DataSourcePluginOptionsEditorProps<KafkaDataSourceOption
 interface State {}
 
 export class ConfigEditor extends PureComponent<Props, State> {
-  onAPIKeyChange = (event: ChangeEvent<HTMLInputElement>) => {
```

Comment:
I did not see apiKey being used anywhere, so I removed it.  Was able to leverage the existing code for it to handle the SASL Password